gammapy/background/make.py

```diff
@@ -0,0 +1,470 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import numpy as np
+from astropy.units import Quantity
+from astropy.coordinates import Angle
+from astropy.table import Table
+from astropy.io import fits
+from ..background import CubeBackgroundModel
+from ..obs import DataStore
+from ..data import EventListDataset
+
+__all__ = ['make_bg_cube_model',
+           'define_cube_binning',
+           'fill_events',
+           'divide_bin_volume',
+           'set_zero_level',
+           'smooth'
+           ]
+
+# TODO: restructure bg models:
+#       - rename CubeBackgroundModel -> CubeModel
+#       - create a CubeBackgroundModel class that contains:
+#          events_cube, livetime_cube, bg_model_cube
+#          methods to produce the bg cube model:
+#           'define_cube_binning',
+#           'fill_events',
+#           'divide_bin_volume',
+#           'set_zero_level',
+#           'smooth'
+# @ C. Deil: does it make sense?
+
```

Comment:
Yes, I think having separate container classes makes sense:
- one that represents a cube
- one that contains the counts, time and rate cubes

I don't have a clear picture yet on whether it would be good to have other classes, e.g.
- Grouping specification
- The algorithm that takes a grouping specification and produces all the cubes

If you can come up with a written down proposal for the hangout on Wednesday that would be good,
otherwise we start by reviewing this PR and then do some brainstorming how to best structure this (within the given time left, ~ 1 week).